checks.d/vsphere.py

```diff
@@ -523,112 +523,181 @@ def get_external_host_tags(self):
+            if isinstance(obj, vim.Folder):
+                # Do not tag with root folder
+                if depth:
+                    tags.append(obj.name)
+
+                for resource in obj.childEntity:
+                    self.pool.apply_async(
+                        browse_mor,
+                        args=(resource, tags, depth + 1)
+                    )
+
+            # Datacenter
+            elif isinstance(obj, vim.Datacenter):
+                tags.append(u"vsphere_datacenter:{0}".format(obj.name))
+
+                for resource in obj.hostFolder.childEntity:
+                    self.pool.apply_async(
+                        browse_mor,
+                        args=(resource, tags, depth + 1)
+                    )
+
+            # ClusterComputeResource
+            elif isinstance(obj, vim.ClusterComputeResource):
+                tags.append(u"vsphere_cluster:{0}".format(obj.name))
+
+                for host in obj.host:
+                    # Skip non-host
+                    if not hasattr(host, 'vm'):
+                        continue
+
+                    self.pool.apply_async(
+                        browse_mor,
+                        args=(host, tags, depth + 1)
+                    )
+
+            # Host
+            elif isinstance(obj, vim.HostSystem):
+                if regexes and regexes.get('host_include') is not None:
+                    match = re.search(regexes['host_include'], obj.name)
+                    if not match:
+                        self.log.debug(
+                            u"Filtered out VM %s because of host_include_only_regex", obj.name
+                        )
+                        return
+
+                watched_mor = dict(
+                    mor_type='host', mor=obj, hostname=obj.name, tags=tags + [u"vsphere_type:host"]
                 )
-
-        elif obj_type == 'datacenter':
-            dc_tag = "vsphere_datacenter:%s" % obj.name
-            tags_copy.append(dc_tag)
-            for compute_resource in obj.hostFolder.childEntity:
-                # Skip non-compute resource
-                if not hasattr(compute_resource, 'host'):
-                    continue
-                self.pool.apply_async(
-                    self._cache_morlist_raw_atomic,
-                    args=(i_key, 'compute_resource', compute_resource, tags_copy, regexes, include_only_marked)
+                self.morlist_raw[instance_key].append(watched_mor)
+
+                tags.append(u"vsphere_host:{}".format(obj.name))
+                for vm in obj.vm:
+                    if vm.runtime.powerState != 'poweredOn':
+                        continue
+                    self.pool.apply_async(
+                        browse_mor,
+                        args=(vm, tags, depth + 1)
+                    )
+
+            # Virtual Machine
+            elif isinstance(obj, vim.VirtualMachine):
+                if regexes and regexes.get('vm_include') is not None:
```

Comment:
shouldn't we use `_is_excluded()` here? I see we're only using it for the tests.